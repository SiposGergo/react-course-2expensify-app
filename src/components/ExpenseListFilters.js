import React from "react";
import { connect } from "react-redux";
import { setTextFilter, sortByAmount, sortByDate, setEndDate, setStartDate } from "../actions/filters";
import { DateRangePicker } from "react-dates";

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDateChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChanged = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e) => {
        if (e.target.value === "date") {
            this.props.sortByDate();
        }
        else if (e.target.value === "amount") {
            this.props.sortByAmount();
        }
    };

    render() {
        return (
            <div>
                <input type="text" defaultValue={this.props.filters.text} onChange={this.onTextChange} />
                <select
                    defaultValue={this.props.filters.sortBy}
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDateChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChanged}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false} />
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch, props) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (date) => dispatch(setStartDate(date)),
    setEndDate: (date) => dispatch(setEndDate(date))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);