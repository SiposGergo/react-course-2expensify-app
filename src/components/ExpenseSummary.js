import React from "react";
import selectTotalExpenses from "../selectors/expenses-total";
import getVisibleExpenses from "../selectors/expenses";
import { connect } from "react-redux";
import numeral from "numeraljs"

export const ExpenseSummary = (props) => {
    const expenseWord = props.expenseCount === 1 ? "expense" : "expenses";
    const formattedTotal = numeral(props.expensesTotal / 100).format("$0,0.00");
    return (
        <div>
            <h1>
                Viewing {props.expenseCount} {expenseWord} totalling {formattedTotal}
            </h1>
        </div>
    )
}

const mapStateToProps = (state, props) => {
    const visibles = getVisibleExpenses(state.expenses, state.filters);
    const total = selectTotalExpenses(visibles);
    return {
        expenseCount: visibles.length,
        expensesTotal: total
    }
}

export default connect(mapStateToProps)(ExpenseSummary)

