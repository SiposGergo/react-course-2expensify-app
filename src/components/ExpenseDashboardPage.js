import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters.js";
import ExpenseSummary from "./ExpenseSummary";

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;