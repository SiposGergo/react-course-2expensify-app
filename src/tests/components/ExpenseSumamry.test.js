import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import React from "react";

test('should render correctly sumary with one expense', () => {
    const wrapper = shallow(
        <ExpenseSummary expenseCount={1} expensesTotal={1500}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render correctly sumary with multiple expenses', () => {
    const wrapper = shallow(
        <ExpenseSummary expenseCount={23} expensesTotal={454564}/>);
    expect(wrapper).toMatchSnapshot();
});