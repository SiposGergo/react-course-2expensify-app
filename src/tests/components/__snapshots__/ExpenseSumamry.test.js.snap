// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly sumary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $4,545.64
  </h1>
</div>
`;

exports[`should render correctly sumary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $15.00
  </h1>
</div>
`;
